bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/ChooseWorkspaceDialog.java

```diff
@@ -464,6 +470,30 @@ protected Combo createPathCombo(Composite panel) {
 					}
 				}
 				okButton.setEnabled(nonWhitespaceFound);
+
+				File location = new File(characters);
+				String normalisedPath = location.getAbsoluteFile().toPath().normalize().toString();
+				String normalisedPathWithSeperator = normalisedPath + File.separator;
+				if (!characters.equalsIgnoreCase(normalisedPath)
+						&& !characters.equalsIgnoreCase(normalisedPathWithSeperator)) {
+				label.setText(
+						IDEWorkbenchMessages.ChooseWorkspaceDialog_ResolvedAbsolutePath0
+								+ location.getAbsoluteFile().toPath().normalize());
+				}
+				else {
+					label.setText(""); //$NON-NLS-1$
+				}
+				if (normalisedPath.startsWith("~")) //$NON-NLS-1$
+				{
+					label.setText(IDEWorkbenchMessages.ChooseWorkspaceDialog_TildeNonExpandedWarning0
+							+ IDEWorkbenchMessages.ChooseWorkspaceDialog_ResolvedAbsolutePath0
+							+ location.getAbsoluteFile().toPath().normalize());
+
+				}
+
+
```

Comment:
Can we get rid of some whitespaces here?

Reply:
Yes. I have removed the whitespaces.